File: frontend/src/constants/apiEndpoints.js
/**
 * Centralized API Endpoints for the Frontend Application.
 * Prevents typos, avoids magic strings, and ensures single source of truth for REST routes.
 */

export const API_BASE_URL = '/api';

export const API_ENDPOINTS = {
  // Auth
  AUTH: {
    TOKEN: `${API_BASE_URL}/auth/token`,
    SIGNUP: `${API_BASE_URL}/auth/signup`,
    SIGNIN: `${API_BASE_URL}/auth/signin`,
    OAUTH_CONFIG: `${API_BASE_URL}/auth/oauth/config`,
    VERIFY_EMAIL: (token) => `${API_BASE_URL}/auth/verify-email?token=${encodeURIComponent(token)}`,
    RESEND_VERIFICATION: `${API_BASE_URL}/auth/resend-verification`,
    OAUTH_GOOGLE: `${API_BASE_URL}/auth/oauth/google`,
    OAUTH_GITHUB: `${API_BASE_URL}/auth/oauth/github`,
    OAUTH_GITHUB_CALLBACK: `${API_BASE_URL}/auth/oauth/github/callback`,
    OAUTH_GOOGLE_CALLBACK: `${API_BASE_URL}/auth/oauth/google/callback`,
    ME: `${API_BASE_URL}/auth/me`,
  },

  // Repositories
  REPOSITORIES: {
    BASE: `${API_BASE_URL}/repositories`,
    BY_ID: (id) => `${API_BASE_URL}/repositories/${id}`,
    TRIGGER_INDEX: (id) => `${API_BASE_URL}/repositories/${id}/index`,
    STATUS: (id) => `${API_BASE_URL}/repositories/${id}/status`,
    BRANCHES: (id) => `${API_BASE_URL}/repositories/${id}/branches`,
    DIFF: (id, base, target) => `${API_BASE_URL}/repositories/${id}/diff?base=${encodeURIComponent(base || 'main')}&target=${encodeURIComponent(target || 'HEAD')}`,
  },

  // Files & Codebase Graph
  FILES: {
    LIST_BY_REPO: (repoId) => `${API_BASE_URL}/files/${repoId}`,
    CONTENT: (repoId) => `${API_BASE_URL}/files/${repoId}/content`,
    GRAPH: (repoId) => `${API_BASE_URL}/files/${repoId}/graph`,
  },

  // Agent Chat & Historical Conversations
  CHAT: {
    SEND: `${API_BASE_URL}/chat`,
    STREAM: `${API_BASE_URL}/chat/stream`,
    CONVERSATIONS_BY_REPO: (repoId) => `${API_BASE_URL}/chat/conversations/${repoId}`,
    CONVERSATION_BY_ID: (conversationId) => `${API_BASE_URL}/chat/conversations/id/${conversationId}`,
    MESSAGES_BY_CONVERSATION: (conversationId) => `${API_BASE_URL}/chat/conversations/${conversationId}/messages`,
  },

  // Specialist Analysis & Audits
  ANALYSIS: {
    RUN: `${API_BASE_URL}/analysis/run`,
    FINDINGS_BY_REPO: (repoId) => `${API_BASE_URL}/analysis/${repoId}`,
  },
};

export const PROVIDERS = {
  GEMINI: 'gemini',
  OPENAI: 'openai',
  ANTHROPIC: 'anthropic',
  GROQ: 'groq',
  DEEPSEEK: 'deepseek',
  OLLAMA: 'ollama',
};

export const VIEW_MODES = {
  CHAT: 'chat',
  CODE: 'code',
  GRAPH: 'graph',
  FINDINGS: 'findings',
};

File: backend/src/main/java/com/aicodebase/architect/constant/ApiEndpoints.java
package com.aicodebase.architect.constant;

public final class ApiEndpoints {

    private ApiEndpoints() {
        // Prevent instantiation
    }

    // Base API prefixes
    public static final String API_BASE = "/api";

    // Auth endpoints
    public static final String AUTH = API_BASE + "/auth";
    public static final String AUTH_TOKEN = "/token";

    // Repository endpoints
    public static final String REPOSITORIES = API_BASE + "/repositories";
    public static final String REPOSITORY_BY_ID = "/{id}";
    public static final String REPOSITORY_INDEX = "/{id}/index";
    public static final String REPOSITORY_STATUS = "/{id}/status";
    public static final String REPOSITORY_BRANCHES = "/{id}/branches";
    public static final String REPOSITORY_DIFF = "/{id}/diff";

    // Chat endpoints
    public static final String CHAT = API_BASE + "/chat";
    public static final String CHAT_STREAM = "/stream";
    public static final String CHAT_CONVERSATIONS = "/conversations/{repositoryId}";
    public static final String CHAT_CONVERSATION_BY_ID = "/conversations/id/{conversationId}";
    public static final String CHAT_MESSAGES = "/conversations/{conversationId}/messages";

    // Analysis & audit endpoints
    public static final String ANALYSIS = API_BASE + "/analysis";
    public static final String ANALYSIS_RUN = "/run";
    public static final String ANALYSIS_BY_REPO = "/{repositoryId}";

    // File explorer & codebase graph endpoints
    public static final String FILES = API_BASE + "/files";
    public static final String FILES_BY_REPO = "/{repositoryId}";
    public static final String FILE_CONTENT = "/{repositoryId}/content";
    public static final String FILE_GRAPH = "/{repositoryId}/graph";

    // AI Service WebClient route constants
    public static final class AiService {
        private AiService() {}

        public static final String INDEX = "/api/index";
        public static final String INDEX_STATUS = "/api/index/status/{repositoryId}";
        public static final String BRANCHES = "/api/index/{repositoryId}/branches";
        public static final String DIFF = "/api/index/{repositoryId}/diff";
        public static final String CHAT = "/api/chat";
        public static final String CHAT_STREAM = "/api/chat/stream";
        public static final String ANALYZE = "/api/analyze";
        public static final String GRAPH = "/api/graph/{repositoryId}";
    }
}
